lib/tbd/psi.rb

```diff
@@ -885,134 +948,116 @@ def derate(model = nil, id = "", s = {}, lc = nil)
+    argh[:floor_ut     ] = 0        unless argh.key?(:floor_ut     )
+    argh[:wall_option  ] = ""       unless argh.key?(:wall_option  )
+    argh[:roof_option  ] = ""       unless argh.key?(:roof_option  )
+    argh[:floor_option ] = ""       unless argh.key?(:floor_option )
+    argh[:gen_ua       ] = false    unless argh.key?(:gen_ua       )
+    argh[:ua_ref       ] = ""       unless argh.key?(:ua_ref       )
+    argh[:gen_kiva     ] = false    unless argh.key?(:gen_kiva     )
 
     # Create the Topolys Model.
     t_model = Topolys::Model.new
 
     # "true" if any space/zone holds valid setpoint temperatures. With invalid
     # inputs, these 2x methods return "false", ignoring any
     # setpoint-based logic, e.g. semi-heated spaces (DEBUG errors are logged).
-    setpoints = heatingTemperatureSetpoints?(model)
-    setpoints = coolingTemperatureSetpoints?(model) || setpoints
-
-    # "true" if any space/zone is part of an HVAC air loop. With invalid inputs,
-    # the method returns "false", ignoring any air-loop related logic, e.g.
-    # plenum zones as HVAC objects (DEBUG errors are logged).
-    airloops = airLoopsHVAC?(model)
+    heated = heatingTemperatureSetpoints?(model)
+    cooled = coolingTemperatureSetpoints?(model)
+    argh[:setpoints] = heated || cooled
 
     model.getSurfaces.sort_by { |s| s.nameString }.each do |s|
-      # Fetch key attributes of opaque surfaces. Method returns nil with invalid
-      # input (DEBUG and ERROR messages may be logged). TBD ignores them.
-      surface = properties(model, s)
-      next if   surface.nil?
-
-      # Similar to "setpoints?" methods above, the boolean methods below also
-      # return "false" with invalid inputs, ignoring any space/zone
-      # conditioning-based logic (e.g. semi-heated spaces, mislabelling a
-      # plenum as an unconditioned zone).
-      if setpoints
-        if surface[:space].thermalZone.empty?
-          plenum = plenum?(surface[:space], airloops, setpoints)
-          surface[:conditioned] = false unless plenum
-        else
-          zone = surface[:space].thermalZone.get
-          heat = maxHeatScheduledSetpoint(zone)
-          cool = minCoolScheduledSetpoint(zone)
-
-          unless heat[:spt] || cool[:spt]
-            plenum     = plenum?(surface[:space], airloops, setpoints)
-            heat[:spt] = 21                   if plenum
-            cool[:spt] = 24                   if plenum
-            surface[:conditioned] = false unless plenum
-          end
-
-          free = heat[:spt] && heat[:spt] < -40 && cool[:spt] && cool[:spt] > 40
-          surface[:conditioned] = false if free
-        end
-      end
-
-      # Recover if valid setpoints.
-      surface[:heating] = heat[:spt] if heat && heat[:spt]
-      surface[:cooling] = cool[:spt] if cool && cool[:spt]
-
-      tbd[:surfaces][s.nameString] = surface
-    end                                            # (opaque) surfaces populated
```

Comment:
Moving away from _plenum_ status to determine _indirectly-conditioned_ spaces. Logic moved to `geo.rb`.